packages/wallet-connection/test/src/mock-agoric-iframe-messenger.js

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable no-underscore-dangle */
+// @ts-check
+import { LitElement, html } from 'lit';
+
+/**
+ * @typedef {Object} MockIframeController - orchestrates a MockAgoricIframeMessenger
+ * @property {Object} [ref] - The current MockAgoricIframeMessenger instance;
+ */
+
+/**
+ * @param {string} mockController
+ */
+export const makeMockAgoricIframeMessenger = mockController =>
+  class MockAgoricIframeMessenger extends LitElement {
+    static get properties() {
+      return { src: { type: String } };
+    }
+
+    constructor() {
+      super();
+      this.src = '';
+      this.sentData = null;
+      window[mockController].ref = this;
```

Comment:
I tried passing a regular object reference as the controller but it seemed that whenever I modified it it tried to reinstantiate the class and caused an error.

Reply:
I didn't specify very clearly, but I expected that the `makeXXX` functions would only be explicitly called once at the top level of the test, and that the API exported by the different files would not change, only be added to (yes, I know that. the lint rules complain about this, but that can be turned off).

I'll rearrange the code in that way.